src/jit/lsraxarch.cpp

```diff
@@ -2437,6 +2437,21 @@ void LinearScan::TreeNodeInfoInitHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
         info->srcCount += GetOperandSourceCount(intrinsicTree->gtOp.gtOp2);
     }
     info->srcCount += GetOperandSourceCount(intrinsicTree->gtOp.gtOp1);
+
+#ifdef _TARGET_X86_
+    if (intrinsicTree->gtHWIntrinsicId == NI_SSE42_Crc32)
+    {
+        // CRC32 may operate over "byte" but on x86 only RBM_BYTE_REGS could be used as byte registers.
+        //
+        // TODO - currently we use the BaseType to bring the type of the secnod argument
+        // to the code generator. May encode the overload info in other way.
+        var_types srcType = intrinsicTree->gtSIMDBaseType;
+        if (srcType == TYP_UBYTE || srcType == TYP_BYTE)
```

Comment:
Should you use `varTypeIsByte()` to also include TYP_BOOL?

Reply:
Good point! Will do.